Doc/tools/extensions/pyspecific.py

```diff
@@ -311,15 +312,26 @@ class DeprecatedRemoved(Directive):
     final_argument_whitespace = True
     option_spec = {}
 
-    _label = 'Deprecated since version {deprecated}, will be removed in version {removed}'
+    python_version = patchlevel.get_version_info()[0]
+    python_major, python_minor = map(int, python_version.split("."))
+
+    _deprecated_label = 'Deprecated since version {deprecated}, will be removed in version {removed}'
+    _removed_label = 'Deprecated since version {deprecated}, removed in version {removed}'
 
     def run(self):
         node = addnodes.versionmodified()
         node.document = self.state.document
         node['type'] = 'deprecated-removed'
         version = (self.arguments[0], self.arguments[1])
         node['version'] = version
-        label = translators['sphinx'].gettext(self._label)
+        removed_major, removed_minor = map(int, self.arguments[1].split("."))
```

Comment:
I think you can simply do 
```python

        env = self.state.document.settings.env
        current_version = tuple(int(e) for e in env.config.version.split('.'))
        removed_version = tuple(int(e) for e in self.arguments[1].split('.'))
        if current_version < removed_version:
            label = self._deprecated_label
        else:
            label = self._removed_label

        label = translators['sphinx'].gettext(label)
```

Then you won't need the `python_version`, `python_major` and `python_minor` class variables.

Reply:
Oh wow, that's awesome. Did not thought about solving it this way, thanks! I will update the PR.